.github/workflows/python_release.yml

```diff
@@ -0,0 +1,100 @@
+name: Reusable python Release workflow example
+
+on:
+  workflow_call:
+    secrets:
+      GCP_PROJECT_ID:
+        required: true
+      GCP_ARTIFACT_REGISTRY_URI:
+        required: true
+      GCP_DOCKER_REGISTRY:
+        required: true
+      GCP_SA_KEY:
+        required: true
+      GCP_ARTIFACT_REGISTRY_PYTHON_URI:
+        required: true
+
+
+      
+    inputs:
+      runs-on:
+        description: Platform to execute on
+        required: false
+        type: string
+        default: ubuntu-latest
+  
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.9]
+
+    env:
+      PROJECT_ID: ${{ secrets.GCP_PROJECT_ID }}
+      ARTIFACT_REGISTRY: ${{ secrets.GCP_ARTIFACT_REGISTRY_URI }}
+      DOCKER_REGISTRY: ${{ secrets.GCP_DOCKER_REGISTRY }}
+      SA_KEY: ${{ secrets.GCP_SA_KEY }}
+      GIT_TERMINAL_PROMPT: 0
+
+    steps:
+    - uses: actions/checkout@v2
+    
+
+    - id: 'auth'
+      uses: 'google-github-actions/auth@v0'
+      with:
+        credentials_json: '${{ secrets.GCP_SA_KEY }}'
+
+    - name: Set up Cloud SDK
+      uses: 'google-github-actions/setup-gcloud@v0'
+      with:
+        project_id: ${{ secrets.GCP_PROJECT_ID }}
+        #service_account_key: ${{ secrets.GCP_SA_KEY }}
+        #export_default_credentials: true
+    
+    - name: Authenticate with GCR #quiet to not freeze.
+      run: |
+        gcloud auth configure-docker ${{ secrets.GCP_ARTIFACT_REGISTRY_URI }} --quiet
+        gcloud config set artifacts/location europe-west1
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4.2.0
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install dependencies
+      uses: 'hagaltech/common-workflows/.github/actions/poetry-install@main'
+      with:
+        poetry-repository: ${{secrets.GCP_ARTIFACT_REGISTRY_PYTHON_URI}}/$PROJECT_ID/${DOCKER_REGISTRY}
+      
+
+    - name: Check codestyle
+      uses: 'hagaltech/common-workflows/.github/actions/check_python@main'
+
+    - name: Run tests
+      run: |
+        coverage run -m pytest -v tests
+        coverage xml
+
+    - name: Create hash for Docker image
+      run: echo "HASH=$(echo $GITHUB_SHA | cut -c 1-7)" >> $GITHUB_ENV
+
+    - name: Get service name from TOML file
+      run: echo "SERVICE_NAME=$(sed -n -e 's/^.*name\s*=\s* //p' pyproject.toml | tr -d '"')" >> $GITHUB_ENV
```

Comment:
this fails on e.g. https://github.com/hagaltech/ems-cloud-lib/blob/main/pyproject.toml - as it has two fields called "name" (this is the case for any package that builds on our internal packages).